lib/solid_queue/supervisor.rb

```diff
@@ -147,11 +152,11 @@ def reap_terminated_forks
       end
 
       def replace_fork(pid, status)
-        if supervised_fork = forks.delete(pid)
-          SolidQueue.logger.info "[SolidQueue] Restarting fork[#{status.pid}] (status: #{status.exitstatus})"
-          start_fork(supervised_fork)
-        else
-          SolidQueue.logger.info "[SolidQueue] Tried to replace fork[#{pid}] (status: #{status.exitstatus}, fork[#{status.pid}]), but it had already died  (status: #{status.exitstatus})"
+        SolidQueue.instrument(:replace_fork, supervisor_pid: ::Process.pid, pid: pid, status: status) do |payload|
+          if supervised_fork = forks.delete(pid)
+            payload[:fork] = supervised_fork
+            start_fork(supervised_fork)
+          end
         end
       end
```

Comment:
@rosa This "Restarting fork" log line was a key clue in helping us debug and I know just how common it is for servers to auto-restart rails processes because of memory so what if we future-proof this for all the people in the years ahead who are going to be diagnosing auto-killed workers. :) 

In our situation the line previously read as:
```
[SolidQueue] Restarting fork[67] (status: )
```

You're already correcting the "restarting fork" language and I think the new log output for this situation will read as:
```
SolidQueue-#{ver} replace_fork (#{ms}) supervisor_pid: 12345, pid: 67, status:
```

That's slightly more helpful but for someone like me who isn't super familiar with SolidQueue it doesn't lead me much towards a possible root cause. I probably would have still posted an Issue to get help. What if you:

1. Notice if `status.pid` is nil and display `SIGKILL` for that
2. Make it clear that `status` is the status of the previous worker which is being replaced; I wouldn't infer that based on the new log output.
3. Include an additional `msg` key in the hash.

Here's my suggested improvement which incorporates these:
```
SolidQueue-#{ver} replace_fork (#{ms}) supervisor_pid: 12345, pid: 67, status: SIGKILL, msg: "Detected a terminated worker so it's being replaced with a new one. Exit status of previous worker was #{status || 'SIGKILL'}"
```

Reply:
Thanks for taking the time to look into this one! I've added some more info in e5790d093748d2d181dfabc930cbf29174d441bd, which I hope it's useful in these cases.

Reply:
👍 That looks great! Now there are definitely some clear clues as to what’s happening. I’m excited about all the investment in SolidQueue and am leaning in with a new product I’m building (rather than my old Sidekiq + Redis setup). Thanks for all your work on this!